### ISSUE TITLE
Add error when display type not passed to doNotify

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry-redux#30]
## The Issue
Per comments in https://github.com/lbryio/lbry-app/pull/1493#issuecomment-390298790, we should add an error when the doNotify function does not contain all required parameters. If we don't, the notifications get queued up and other notifications don't show afterwards. 

## System Configuration

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

**Comment by jessopb** [comment]
Agreed, I think my socialShare modal commit was good, but I cancelled it because my feature branch was behaving very strangely. The wallet <Address> doNotify button (notification object which has no id) was matching to my MODALS.SOCIAL_SHARE constant which was also undefined causing it to run my component without the rest of its props, crashing the app.

Also, modalRouter checks to see if "notification" exists, but doesn't check to see if notification.id exists before the switch statement switches on 'id'.
